sui/typescript/dapp_kit_example/src/client/components/ShinamiClientProvider.tsx

```diff
@@ -0,0 +1,22 @@
+import { createSuiClient } from "@shinami/clients/sui";
+import { SuiClient } from "@mysten/sui/client";
+import { createContext } from "react";
+import { NODE_SERVICE_PUBLICLY_VISIBLE_TESTNET_ACCESS_KEY } from "../config.js";
+
+export interface ShinamiClientProviderContext {
+    client: SuiClient;
+}
+
+export const ShinamiClientContext = createContext<ShinamiClientProviderContext | null>(null);
+
+export type ShinamiClientProviderProps = {
+    children?: React.ReactNode;
+};
+
+export function ShinamiClientProvider(props: ShinamiClientProviderProps) {
```

Comment:
This provider as is isn't very helpful, because it's using one hard-coded access key. In this case, you might as well just use one global variable (like we do in most places). The advantage of using a context provider is to allow people to instantiate multiple instances and use them in different parts of the component tree. So in that vein, a more helpful version is to expose access key as a prop.